PokeAlarm/Manager.py

```diff
@@ -740,6 +775,113 @@ def process_gym(self, gym):
         for thread in threads:
             thread.join()
 
+    def process_raid(self, raid):
+        # Quick check for enabled
+        if self.__raid_settings['enabled'] is False:
+            log.debug("Raid ignored: notifications are disabled.")
+            return
+
+        id_ = raid['id']
+
+        pkmn_id = raid['pkmn_id']
+        raid_end = raid['expire_time']
+        name = None
+
+        # raid history will contain the end date and also the pokemon if it has hatched
+        if id_ in self.__raid_hist:
+            old_raid_end = self.__raid_hist[id_]['expire_time']
+            old_raid_pkmn = self.__raid_hist[id_].get('pkmn_id', 0)
+            if old_raid_end == raid_end:
+                if old_raid_pkmn == pkmn_id: # raid with same end time exists and it has same pokemon id, skip it
+                    if self.__quiet is False:
+                        log.debug("Raid {} was skipped because it was previously processed.".format(id))
+                    return
+
+        self.__raid_hist[id_] = dict(expire_time=raid_end, pkmn_id=pkmn_id)
+
+        lat, lng = raid['lat'], raid['lng']
+        dist = get_earth_dist([lat, lng], self.__latlng)
+
+        # Check if in geofences
+        if len(self.__geofences) > 0:
+            inside = False
+            for gf in self.__geofences:
+                inside |= gf.contains(lat, lng)
+            if inside is False:
+                if self.__quiet is False:
+                    log.info("Raid update ignored: located outside geofences.")
+                return
+        else:
+            log.debug("Raid inside geofences was not checked because no geofences were set.")
```

Comment:
Consider being consistent with the other process_* functions by using the self.check_geofences(name, lat, lng) function.  This will also allow for the geofence name to be process as a DTS token.